File: src/main/java/de/mikeyllp/miniGamesV4/utils/MessageUtils.java
package de.mikeyllp.miniGamesV4.utils;

import org.bukkit.command.CommandSender;
import org.bukkit.configuration.file.FileConfiguration;
import org.bukkit.configuration.file.YamlConfiguration;
import org.bukkit.plugin.java.JavaPlugin;

import java.io.File;


public class MessageUtils {

    // To get the Config
    private static JavaPlugin plugin;

    private static FileConfiguration getLangConfig(String lang) {
        File file = new File(plugin.getDataFolder(), "languages/" + lang + ".yml");
        return YamlConfiguration.loadConfiguration(file);
    }

    private static FileConfiguration config() {
        return plugin.getConfig();
    }

    private static String PREFIX;

    public static void init(JavaPlugin pl) {
        plugin = pl;
        reloadConfig();
    }

    public static void reloadConfig() {
        plugin.reloadConfig();
        PREFIX = plugin.getConfig().getString("prefix", "").replace("%message%", "");
    }

    public static String prefix() {
        return PREFIX;
    }


    // You can use this very easily because you only set the message and the prefix are automatically added.
    public static void sendCustomMessage(CommandSender sender, String message) {
        sender.sendRichMessage(prefix() + message);
    }

    // The method for the standard warning messages.
    public static void sendCustomWarnMessage(CommandSender sender, String message) {
        sender.sendRichMessage(prefix() + "<red>" + message + "</red>");
    }

    // You can use this to send a help message with the command and description.
    public static void sendHelpMessage(CommandSender sender, String command, String description) {
        sender.sendRichMessage("<white>" + command + "</white> <gray>" + description);
    }

    // Normal messages


    // Info
    public static void sendNeedReloadMessage(CommandSender sender) {
        sendCustomMessage(sender, getLangConfig(config().getString("language")).getString("normal-message.info.reload"));
    }

    public static void sendGameSwitch(CommandSender sender, String game, boolean state) {
        if (state) {
            sendCustomMessage(sender, getLangConfig(config().getString("language")).getString("normal-message.info.enabled-game").replace("%game%", game));
            return;
        }
        sendCustomMessage(sender, getLangConfig(config().getString("language")).getString("normal-message.info.disabled-game").replace("%game%", game));
    }

    public static void needHelpMessage(CommandSender sender) {
        sendCustomMessage(sender, getLangConfig(config().getString("language")).getString("normal-message.info.need-help").replace("%command%", config().getString("command")));
    }

    // Error Messages
    public static void sendNoPermissionMessage(CommandSender sender) {
        sendCustomWarnMessage(sender, getLangConfig(config().getString("language")).getString("warning-message.no-permission"));
    }

    public static void miniGamesDisabledMessage(CommandSender sender) {
        sendCustomWarnMessage(sender, getLangConfig(config().getString("language")).getString("disabled-game"));
    }

    public static void sendAlreadyInGameMessage(CommandSender sender) {
        sendCustomWarnMessage(sender, getLangConfig(config().getString("language")).getString("already-in-game"));
    }

    public static void sendNoOnlinePlayerMessage(CommandSender sender) {
        sendCustomWarnMessage(sender, getLangConfig(config().getString("language")).getString("player-not-found"));
    }

    public static void sendNoInviteYourselfMessage(CommandSender sender) {
        sendCustomWarnMessage(sender, getLangConfig(config().getString("language")).getString("no-invite-yourself"));
    }
}

File: src/main/java/de/mikeyllp/miniGamesV4/utils/MessageUtils.kt
package de.mikeyllp.miniGamesV4.utils

import de.mikeyllp.miniGamesV4.config
import net.kyori.adventure.text.Component
import net.kyori.adventure.text.minimessage.MiniMessage
import net.kyori.adventure.text.minimessage.tag.Tag
import net.kyori.adventure.text.minimessage.tag.resolver.TagResolver
import org.bukkit.command.CommandSender


object MessageUtils {

    lateinit var prefix: Component


    fun initCustomTags() {
        TagResolver.resolver("prefix") { queue, context ->
            Tag.inserting(prefix)
        }
    }


    fun sendMessage(sender: CommandSender, message: String) {

        sender.sendMessage(MiniMessage.miniMessage().deserialize(message))
    }

    fun sendNeedReloadMessage(sender: CommandSender) {
        sendMessage(
            sender,
            translatable("normal-message.info.reload")
        )
    }

    fun sendGameSwitch(sender: CommandSender, game: String, state: Boolean) {
        val key = if (state) "enabled-game" else "disabled-game"

        sendMessage(
            sender, translatable("normal-message.info.$key", game)
        )
    }

    fun needHelpMessage(sender: CommandSender) {
        val command = config.getString("command") ?: "minigames"
        sendMessage(
            sender,
            translatable("normal-message.info.need-help", command)
        )
    }


    fun sendNoPermissionMessage(sender: CommandSender) {
        sendMessage(
            sender,
            translatable("warning-message.no-permission")
        )
    }

    fun miniGamesDisabledMessage(sender: CommandSender) {
        sendMessage(
            sender,
            translatable("warning-message.disabled-game")
        )
    }

}
